app/main.py

```diff
@@ -1,8 +1,23 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    new_list = []
+    for person in people:
+        new_person = Person(person["name"], person["age"])
+        new_list.append(new_person)
+
+    for other_person in people:
+        instance = Person.people[other_person["name"]]
+        if "wife" in other_person and other_person["wife"]:
+            instance.wife = Person.people[other_person["wife"]]
+        elif "husband" in other_person and other_person["husband"]:
+            instance.husband = Person.people[other_person["husband"]]
```

Comment:
The code assumes that if a person has a 'wife' or 'husband', that person is already in the `Person.people` dictionary. This could lead to a `KeyError` if the spouse is not in the list of people processed by `create_person_list`. Consider adding error handling or ensuring all referenced people are included in the initial list.